File: tests/e2e/test_e2e.py
"""E2E test for the Python SDK against a live recorder server."""

import os
import sys
import time

sys.path.insert(0, os.path.dirname(__file__))
from thea.client import RecorderClient


def main():
    url = os.environ.get("THEA_URL", "http://localhost:9123")
    client = RecorderClient(url)

    print("[python] Waiting for server...")
    client.wait_until_ready(timeout=30)

    print("[python] Starting display...")
    client.start_display()

    print("[python] Health check...")
    health = client.health()
    assert health["status"] == "ok", f"expected ok, got {health['status']}"
    print(f"[python] Health: status={health['status']} display={health.get('display')}")

    print("[python] Adding panel...")
    client.add_panel("editor", "Code Editor", 80)

    print("[python] Updating panel...")
    client.update_panel("editor", "print('hello from Python')", focus_line=1)

    print("[python] Listing panels...")
    panels = client.list_panels()
    assert len(panels) == 1, f"expected 1 panel, got {len(panels)}"
    assert panels[0]["name"] == "editor"

    print("[python] Starting recording...")
    client.start_recording("python-e2e-test")

    time.sleep(2)

    print("[python] Checking recording status...")
    status = client.recording_status()
    assert status["recording"] is True, f"expected recording=True"

    print("[python] Stopping recording...")
    result = client.stop_recording()
    assert result["path"], "expected non-empty path"
    print(f"[python] Recording saved: {result['path']} ({result['elapsed']:.1f}s)")

    print("[python] Removing panel...")
    client.remove_panel("editor")

    print("[python] Listing recordings...")
    recordings = client.list_recordings()
    assert len(recordings) >= 1, f"expected >= 1 recording"

    print("[python] Stopping display...")
    client.stop_display()

    print("[python] Cleanup...")
    client.cleanup()

    print("[python] ALL PASSED")


if __name__ == "__main__":
    main()

File: tests/e2e/TestE2E.java
import com.recorder.RecorderClient;
import java.time.Duration;

/**
 * E2E test for the Java SDK against a live recorder server.
 */
public class TestE2E {
    public static void main(String[] args) throws Exception {
        String url = System.getenv("THEA_URL");
        if (url == null || url.isEmpty()) url = "http://localhost:9123";

        try (var client = new RecorderClient(url)) {
            System.out.println("[java] Waiting for server...");
            client.waitUntilReady(Duration.ofSeconds(30));

            System.out.println("[java] Starting display...");
            client.startDisplay();

            System.out.println("[java] Health check...");
            var health = client.health();
            assertEq("ok", health.status(), "health.status");
            System.out.printf("[java] Health: status=%s display=%s%n", health.status(), health.display());

            System.out.println("[java] Adding panel...");
            client.addPanel("editor", "Code Editor", 80);

            System.out.println("[java] Updating panel...");
            client.updatePanel("editor", "System.out.println(\"hello from Java\")", 1);

            System.out.println("[java] Listing panels...");
            var panels = client.listPanels();
            assertEq(1, panels.size(), "panels.size");
            assertEq("editor", panels.get(0).name(), "panel name");

            System.out.println("[java] Starting recording...");
            client.startRecording("java-e2e-test");

            Thread.sleep(2000);

            System.out.println("[java] Checking recording status...");
            var status = client.recordingStatus();
            assertTrue(status.recording(), "expected recording=true");

            System.out.println("[java] Stopping recording...");
            var result = client.stopRecording();
            assertTrue(result.path() != null && !result.path().isEmpty(), "expected non-empty path");
            System.out.printf("[java] Recording saved: %s (%.1fs)%n", result.path(), result.elapsed());

            System.out.println("[java] Removing panel...");
            client.removePanel("editor");

            System.out.println("[java] Listing recordings...");
            var recordings = client.listRecordings();
            assertTrue(recordings.size() >= 1, "expected >= 1 recording");

            System.out.println("[java] Stopping display...");
            client.stopDisplay();

            System.out.println("[java] Cleanup...");
            client.cleanup();

            System.out.println("[java] ALL PASSED");
        }
    }

    static void assertEq(Object expected, Object actual, String label) {
        if (!expected.equals(actual)) {
            System.err.printf("[java] ASSERTION FAILED: %s: expected %s, got %s%n", label, expected, actual);
            System.exit(1);
        }
    }

    static void assertTrue(boolean cond, String msg) {
        if (!cond) {
            System.err.printf("[java] ASSERTION FAILED: %s%n", msg);
            System.exit(1);
        }
    }
}
